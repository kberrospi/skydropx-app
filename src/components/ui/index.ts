export * from './Formulario';
export * from './TablePrice';
