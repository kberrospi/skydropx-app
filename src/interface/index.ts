export * from './shipments'
export * from './dataForm'
export * from './dataPost'
export * from './labels'