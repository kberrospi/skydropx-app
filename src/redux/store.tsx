import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import { labelSlice, shipmentSlice } from './reducers'

export function makeStore() {
  return configureStore({
    reducer: { 
      shipment: shipmentSlice.reducer,
      label: labelSlice.reducer
    },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store