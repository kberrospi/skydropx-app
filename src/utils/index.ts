export * from './data-post'
export * from './helperForm'
export * from './helperLabels'