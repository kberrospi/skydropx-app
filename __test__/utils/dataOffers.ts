import { Included, Type } from "../../src/interface"

export const offer: Included[] = [
  {
      id: "376092",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:42.789-05:00",
          updated_at: "2022-04-12T09:21:48.914-05:00",
          amount_local: "155.0",
          currency_local: "MXN",
          provider: "FEDEX",
          service_level_name: "Fedex Express Saver",
          service_level_code: "FEDEX_EXPRESS_SAVER",
          service_level_terms: null,
          days: 5,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: false,
          out_of_area_pricing: "0.0",
          total_pricing: "155.0",
          is_ocurre: false
      }
  },
  {
      id: "376093",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:42.793-05:00",
          updated_at: "2022-04-12T09:21:48.921-05:00",
          amount_local: "191.0",
          currency_local: "MXN",
          provider: "FEDEX",
          service_level_name: "Standard Overnight",
          service_level_code: "STANDARD_OVERNIGHT",
          service_level_terms: null,
          days: 2,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: false,
          out_of_area_pricing: "0.0",
          total_pricing: "191.0",
          is_ocurre: false
      }
  },
  {
      id: "376096",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:48.635-05:00",
          updated_at: "2022-04-12T09:21:48.635-05:00",
          amount_local: "93.0",
          currency_local: "MXN",
          provider: "REDPACK",
          service_level_name: "Ecoexpress",
          service_level_code: "ECOEXPRESS",
          service_level_terms: null,
          days: 5,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: true,
          out_of_area_pricing: "141.52",
          total_pricing: "234.52",
          is_ocurre: false
      }
  },
  {
      id: "376097",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:48.640-05:00",
          updated_at: "2022-04-12T09:21:48.640-05:00",
          amount_local: "145.0",
          currency_local: "MXN",
          provider: "REDPACK",
          service_level_name: "Express",
          service_level_code: "EXPRESS",
          service_level_terms: null,
          days: 2,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: true,
          out_of_area_pricing: "174.36",
          total_pricing: "319.36",
          is_ocurre: false
      }
  },
  {
      id: "376100",
      type: Type.Rates,
      attributes: {
          created_at: "2022-04-12T09:21:48.877-05:00",
          updated_at: "2022-04-12T09:21:48.971-05:00",
          amount_local: "130.0",
          currency_local: "MXN",
          provider: "CARSSA",
          service_level_name: "Nacional",
          service_level_code: "NACIONAL",
          service_level_terms: null,
          days: 10,
          duration_terms: null,
          zone: null,
          arrives_by: null,
          out_of_area: false,
          out_of_area_pricing: "0.0",
          total_pricing: "130.0",
          is_ocurre: false
      }
  }
]