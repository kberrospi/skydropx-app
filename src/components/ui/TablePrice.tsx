import { FC, useState } from 'react';
import { Button, Grid, Loading, Table, Text, Tooltip } from '@nextui-org/react';
import { Included } from '../../interface';
import styles from '../../styles/TablePrice.module.css'
import { creatingLabel } from '../../utils';
import { useAppDispatch, useAppSelector } from '../../redux/hook';


interface Props {
  offers: Included[]
}

export const TablePrice: FC<Props> = ({ offers }) => {

  const [offerSelected, setOfferSelected] = useState<number>(0);
  const { status: statusLabel, data: dataLabel } = useAppSelector((state) => state.label);
  const dispatch = useAppDispatch();

  /* Any por un error aparentemene de la libreria de nextUI */
  const rowSelected = (row: any ) =>{

    setOfferSelected( parseInt( row.currentKey ) )

  }

  return (

    <Grid className={ styles.container }>
      <Text h5 className={ styles['title-table'] }> 
        Elije el servicio de paqueteria que prefieras:  
      </Text>
      <Table 
        className={ styles.table }
        aria-label="tablePrice"
        selectionMode="single"
        onSelectionChange={ rowSelected }
      >
        <Table.Header>
          <Table.Column align='center' > Compañia </Table.Column>
          <Table.Column align='center' > Llega en </Table.Column>
          <Table.Column align='center' > Precio </Table.Column>
        </Table.Header>
        <Table.Body items={offers}  >
          {
            offers.map(( offer ) => (
              
              <Table.Row key={offer.id}  >
                <Table.Cell css={{ textAlign: 'center' }} >
                  <Text className={ styles.text }>
                    {offer.attributes.provider}
                  </Text>
                </Table.Cell>
                <Table.Cell css={{ textAlign: 'center' }} >
                  <Text className={ styles.text }>
                    {offer.attributes.days} días
                  </Text>
                </Table.Cell>
                <Table.Cell css={{ display:'flex', justifyContent:'center'}} >
                  {
                    offer.attributes.cheaper 
                    ?
                    <Tooltip content="Más económico" color='success'>
                      <Text b className={ styles.text && styles.cheaper } >
                        {`${offer.attributes.total_pricing} (${offer.attributes.currency_local}) `}  
                      </Text>
                    </Tooltip>
                    :
                    <Text b className={ styles.text } css={{ padding:'0.2rem 0.5rem' }} >
                      {`${offer.attributes.total_pricing} (${offer.attributes.currency_local}) `}  
                    </Text>
                  }
                </Table.Cell>
              </Table.Row>
              
            ))
          }
        </Table.Body>
      </Table>
      <Button 
        css={{ width:'100%', marginTop:'1rem' }}
        onClick={()=> creatingLabel(offerSelected, dispatch)} 
        shadow
        className={ statusLabel === 'loading' ? '' : styles.button }
        disabled={ statusLabel === 'loading' }
      > 
         { statusLabel === 'loading' 
          ? <Loading  color="currentColor" size="sm" />
          : 'Enviar'
        } 
      </Button>
    </Grid>
  )
}
