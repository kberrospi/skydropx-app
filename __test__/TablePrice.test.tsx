import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../src/redux/store";
import { TablePrice } from '../src/components/ui/TablePrice';
import { offer } from "./utils/dataOffers";
import { creatingLabel } from "../src/utils";


const state = store.getState();
store.dispatch = jest.fn();


jest.mock( '../src/utils', ()=> {
  creatingLabel: jest.fn()
} )

describe('Pruebas en TablePrice', () => { 

  const wrapper = render(
    <Provider store={store}>
     <TablePrice offers={ offer } />
    </Provider>
  )

  test('Debe renderiza la tabla correctamente', () => { 
    expect(wrapper).toMatchSnapshot()
  })

  /* test('Debe llamar el metodo creatingLabel', () => { 
    fireEvent.click(wrapper.getByRole('button'))
    expect(creatingLabel).toHaveBeenCalledTimes(1)
  }) */

})