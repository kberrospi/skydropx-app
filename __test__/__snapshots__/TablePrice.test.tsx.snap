// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TablePrice Debe renderiza la tabla correctamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item container"
      >
        <h5
          class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css title-table"
        >
          Elije el servicio de paqueteria que prefieras:
        </h5>
        <div
          class="nextui-c-hSfoSF nextui-c-hSfoSF-gikTHb-shadow-true nextui-c-hSfoSF-bzEnZV-borderWeight-normal nextui-c-hSfoSF-iPJLV-css nextui-table-container"
        >
          <table
            aria-describedby=""
            aria-label="tablePrice"
            class="nextui-c-hLONex nextui-c-hLONex-dicOIe-color-primary nextui-c-hLONex-fPgLZC-lineWeight-light nextui-c-hLONex-dRJGNh-hoverable-true table"
            id="react-aria2107865249-2"
            role="grid"
            tabindex="0"
          >
            <thead
              class="nextui-c-PJLV nextui-table-row-group"
              role="rowgroup"
            >
              <tr
                class="nextui-c-fRbcxI nextui-table-header-row"
                role="row"
              >
                <th
                  class="nextui-c-gkciFS nextui-c-PJLV nextui-c-gkciFS-gjdJOs-align-center nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                  data-key=".0.0"
                  id="react-aria2107865249-2-.0.0"
                  role="columnheader"
                  tabindex="-1"
                >
                   Compañia 
                </th>
                <th
                  class="nextui-c-gkciFS nextui-c-PJLV nextui-c-gkciFS-gjdJOs-align-center nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                  data-key=".0.1"
                  id="react-aria2107865249-2-.0.1"
                  role="columnheader"
                  tabindex="-1"
                >
                   Llega en 
                </th>
                <th
                  class="nextui-c-gkciFS nextui-c-PJLV nextui-c-gkciFS-gjdJOs-align-center nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                  data-key=".0.2"
                  id="react-aria2107865249-2-.0.2"
                  role="columnheader"
                  tabindex="-1"
                >
                   Precio 
                </th>
              </tr>
              <tr
                aria-hidden="true"
                class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-icCJVMZ-css nextui-table-hidden-row"
              />
            </thead>
            <tbody
              class="nextui-c-PJLV nextui-c-PJLV-iccTTUn-css nextui-table-body"
              role="rowgroup"
            >
              <tr
                aria-hidden="false"
                aria-labelledby="react-aria2107865249-2-376092-.0.0"
                aria-selected="false"
                class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                data-key="376092"
                role="row"
                tabindex="-1"
              >
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376092.0"
                  id="react-aria2107865249-2-376092-.0.0"
                  role="rowheader"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    FEDEX
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376092.1"
                  role="gridcell"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    5
                     días
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                  data-key="376092.2"
                  role="gridcell"
                  tabindex="-1"
                >
                  <b
                    class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                  >
                    155.0 (MXN) 
                  </b>
                </td>
              </tr>
              <tr
                aria-hidden="false"
                aria-labelledby="react-aria2107865249-2-376093-.0.0"
                aria-selected="false"
                class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                data-key="376093"
                role="row"
                tabindex="-1"
              >
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376093.0"
                  id="react-aria2107865249-2-376093-.0.0"
                  role="rowheader"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    FEDEX
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376093.1"
                  role="gridcell"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    2
                     días
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                  data-key="376093.2"
                  role="gridcell"
                  tabindex="-1"
                >
                  <b
                    class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                  >
                    191.0 (MXN) 
                  </b>
                </td>
              </tr>
              <tr
                aria-hidden="false"
                aria-labelledby="react-aria2107865249-2-376096-.0.0"
                aria-selected="false"
                class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                data-key="376096"
                role="row"
                tabindex="-1"
              >
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376096.0"
                  id="react-aria2107865249-2-376096-.0.0"
                  role="rowheader"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    REDPACK
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376096.1"
                  role="gridcell"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    5
                     días
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                  data-key="376096.2"
                  role="gridcell"
                  tabindex="-1"
                >
                  <b
                    class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                  >
                    234.52 (MXN) 
                  </b>
                </td>
              </tr>
              <tr
                aria-hidden="false"
                aria-labelledby="react-aria2107865249-2-376097-.0.0"
                aria-selected="false"
                class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                data-key="376097"
                role="row"
                tabindex="-1"
              >
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376097.0"
                  id="react-aria2107865249-2-376097-.0.0"
                  role="rowheader"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    REDPACK
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376097.1"
                  role="gridcell"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    2
                     días
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                  data-key="376097.2"
                  role="gridcell"
                  tabindex="-1"
                >
                  <b
                    class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                  >
                    319.36 (MXN) 
                  </b>
                </td>
              </tr>
              <tr
                aria-hidden="false"
                aria-labelledby="react-aria2107865249-2-376100-.0.0"
                aria-selected="false"
                class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
                data-key="376100"
                role="row"
                tabindex="-1"
              >
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376100.0"
                  id="react-aria2107865249-2-376100-.0.0"
                  role="rowheader"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    CARSSA
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                  data-key="376100.1"
                  role="gridcell"
                  tabindex="-1"
                >
                  <p
                    class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                  >
                    10
                     días
                  </p>
                </td>
                <td
                  class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                  data-key="376100.2"
                  role="gridcell"
                  tabindex="-1"
                >
                  <b
                    class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                  >
                    130.0 (MXN) 
                  </b>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <button
          class="nextui-c-kZXRhs nextui-c-PJLV nextui-c-kZXRhs-ikltse-color-default nextui-c-kZXRhs-bWGFbV-size-md nextui-c-kZXRhs-cwXrJp-borderWeight-normal nextui-c-kZXRhs-fzQovo-shadow-true nextui-c-kZXRhs-jsIPTU-cv nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-kZXRhs-ihihFAR-css nextui-button nextui-button--ready button"
          data-state="ready"
          type="button"
        >
          <span
            class="nextui-button-text"
          >
            Enviar
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nextui-c-kRHeuF nextui-c-kRHeuF-igNCIse-css nextui-grid-item container"
    >
      <h5
        class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css title-table"
      >
        Elije el servicio de paqueteria que prefieras:
      </h5>
      <div
        class="nextui-c-hSfoSF nextui-c-hSfoSF-gikTHb-shadow-true nextui-c-hSfoSF-bzEnZV-borderWeight-normal nextui-c-hSfoSF-iPJLV-css nextui-table-container"
      >
        <table
          aria-describedby=""
          aria-label="tablePrice"
          class="nextui-c-hLONex nextui-c-hLONex-dicOIe-color-primary nextui-c-hLONex-fPgLZC-lineWeight-light nextui-c-hLONex-dRJGNh-hoverable-true table"
          id="react-aria2107865249-2"
          role="grid"
          tabindex="0"
        >
          <thead
            class="nextui-c-PJLV nextui-table-row-group"
            role="rowgroup"
          >
            <tr
              class="nextui-c-fRbcxI nextui-table-header-row"
              role="row"
            >
              <th
                class="nextui-c-gkciFS nextui-c-PJLV nextui-c-gkciFS-gjdJOs-align-center nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                data-key=".0.0"
                id="react-aria2107865249-2-.0.0"
                role="columnheader"
                tabindex="-1"
              >
                 Compañia 
              </th>
              <th
                class="nextui-c-gkciFS nextui-c-PJLV nextui-c-gkciFS-gjdJOs-align-center nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                data-key=".0.1"
                id="react-aria2107865249-2-.0.1"
                role="columnheader"
                tabindex="-1"
              >
                 Llega en 
              </th>
              <th
                class="nextui-c-gkciFS nextui-c-PJLV nextui-c-gkciFS-gjdJOs-align-center nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
                data-key=".0.2"
                id="react-aria2107865249-2-.0.2"
                role="columnheader"
                tabindex="-1"
              >
                 Precio 
              </th>
            </tr>
            <tr
              aria-hidden="true"
              class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-icCJVMZ-css nextui-table-hidden-row"
            />
          </thead>
          <tbody
            class="nextui-c-PJLV nextui-c-PJLV-iccTTUn-css nextui-table-body"
            role="rowgroup"
          >
            <tr
              aria-hidden="false"
              aria-labelledby="react-aria2107865249-2-376092-.0.0"
              aria-selected="false"
              class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
              data-key="376092"
              role="row"
              tabindex="-1"
            >
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376092.0"
                id="react-aria2107865249-2-376092-.0.0"
                role="rowheader"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  FEDEX
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376092.1"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  5
                   días
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                data-key="376092.2"
                role="gridcell"
                tabindex="-1"
              >
                <b
                  class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                >
                  155.0 (MXN) 
                </b>
              </td>
            </tr>
            <tr
              aria-hidden="false"
              aria-labelledby="react-aria2107865249-2-376093-.0.0"
              aria-selected="false"
              class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
              data-key="376093"
              role="row"
              tabindex="-1"
            >
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376093.0"
                id="react-aria2107865249-2-376093-.0.0"
                role="rowheader"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  FEDEX
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376093.1"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  2
                   días
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                data-key="376093.2"
                role="gridcell"
                tabindex="-1"
              >
                <b
                  class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                >
                  191.0 (MXN) 
                </b>
              </td>
            </tr>
            <tr
              aria-hidden="false"
              aria-labelledby="react-aria2107865249-2-376096-.0.0"
              aria-selected="false"
              class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
              data-key="376096"
              role="row"
              tabindex="-1"
            >
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376096.0"
                id="react-aria2107865249-2-376096-.0.0"
                role="rowheader"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  REDPACK
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376096.1"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  5
                   días
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                data-key="376096.2"
                role="gridcell"
                tabindex="-1"
              >
                <b
                  class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                >
                  234.52 (MXN) 
                </b>
              </td>
            </tr>
            <tr
              aria-hidden="false"
              aria-labelledby="react-aria2107865249-2-376097-.0.0"
              aria-selected="false"
              class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
              data-key="376097"
              role="row"
              tabindex="-1"
            >
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376097.0"
                id="react-aria2107865249-2-376097-.0.0"
                role="rowheader"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  REDPACK
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376097.1"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  2
                   días
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                data-key="376097.2"
                role="gridcell"
                tabindex="-1"
              >
                <b
                  class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                >
                  319.36 (MXN) 
                </b>
              </td>
            </tr>
            <tr
              aria-hidden="false"
              aria-labelledby="react-aria2107865249-2-376100-.0.0"
              aria-selected="false"
              class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
              data-key="376100"
              role="row"
              tabindex="-1"
            >
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376100.0"
                id="react-aria2107865249-2-376100-.0.0"
                role="rowheader"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  CARSSA
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-igjdJOs-css nextui-table-cell"
                data-key="376100.1"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="nextui-c-PJLV nextui-c-PJLV-ijXuRFq-css text"
                >
                  10
                   días
                </p>
              </td>
              <td
                class="nextui-c-PJLV nextui-c-eXPDRa nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-PJLV-ikqvkfk-css nextui-table-cell"
                data-key="376100.2"
                role="gridcell"
                tabindex="-1"
              >
                <b
                  class="nextui-c-PJLV nextui-c-PJLV-idOgsxa-css text"
                >
                  130.0 (MXN) 
                </b>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <button
        class="nextui-c-kZXRhs nextui-c-PJLV nextui-c-kZXRhs-ikltse-color-default nextui-c-kZXRhs-bWGFbV-size-md nextui-c-kZXRhs-cwXrJp-borderWeight-normal nextui-c-kZXRhs-fzQovo-shadow-true nextui-c-kZXRhs-jsIPTU-cv nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-kZXRhs-ihihFAR-css nextui-button nextui-button--ready button"
        data-state="ready"
        type="button"
      >
        <span
          class="nextui-button-text"
        >
          Enviar
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
